cats/configure.py

```diff
@@ -76,26 +76,34 @@ def config_from_file(configpath="") -> Mapping[str, Any]:
     if configpath:
         # if path to config file provided, it is used
         with open(configpath, "r") as f:
-            return yaml.safe_load(f)
+            conf_dict = yaml.safe_load(f)
         logging.info(f"Using provided config file: {configpath}\n")
     else:
         # if no path provided, try to use config environment variable
         cfile = os.getenv("CATS_CONFIG_FILE")
         if cfile is not None:
             try:
                 with open(cfile, "r") as f:
-                    return yaml.safe_load(f)
-                logging.info("Using config.yml found in CATS_CONFIG_FILE\n")
+                    conf_dict = yaml.safe_load(f)
+                logging.info(f"Using {cfile} found in CATS_CONFIG_FILE\n")
             except FileNotFoundError:
                 logging.warning("CATS_CONFIG_FILE config file not found")
-        # if no path provided and no env variable, look for `config.yml` in current directory
-        try:
-            with open("config.yml", "r") as f:
-                return yaml.safe_load(f)
-            logging.info("Using config.yml found in current directory\n")
-        except FileNotFoundError:
-            logging.warning("config file not found")
-            return {}
+        else:
+            # if no path provided and no env variable, look for a file in current directory
+            # we support several file names but warn for deprecated names
+            config_file_names = ["cats_config.yml", "cats_config.yaml", "config.yaml"]
+            cfile = next((x for x in config_file_names if os.path.isfile(x)), "config.yml")
```

Comment:
Why don't we check that the 'config.yml' case is a file, like with the other three? This seems simpler and more comprehensible:

```suggestion
            config_file_names = ["cats_config.yml", "cats_config.yaml", "config.yaml", "config.yml"]
            cfile = next(x for x in config_file_names if os.path.isfile(x))
```

Reply:
The reason I did it like this that we need something in that cfile variable (even if the file doesn't exist) so we can report that it doesn't exist. Otherwise we would need an extra test for None.

Reply:
Ah OK that makes sense - you could potentially add in a comment to say as much so others don't wonder the same thing, but it is not a blocker. I'll resolve this conversation. I am happy and so is Abhishek so please merge when ready.